src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -687,7 +687,13 @@ public static bool IsEnumType(this ITypeSymbol type, [NotNullWhen(true)] out INa
             var hasPrivateField = false;
             foreach (var member in type.GetMembers())
             {
-                if (!(member is IFieldSymbol fieldSymbol))
+                // TODO: Events?
+#if CODE_STYLE
+                var fieldSymbol = member as IFieldSymbol;
+#else
+                var fieldSymbol = member as IFieldSymbol ?? (member as IPropertySymbol)?.AssociatedField;
+#endif
```

Comment:
It looks like there's an existing test hole again that would require IEventSymbol.AssociatedField to be exposed.